File: java/ElectronicSensor.java
// Child class — inherits from IndustrialAsset — Part 1 & 2
public class ElectronicSensor extends IndustrialAsset {

    private double voltage;
    private double currentReading;

    public ElectronicSensor(String name, String serialNumber, double baseValue, double voltage, double currentReading) {
        super(name, serialNumber, baseValue);
        this.voltage = voltage;
        this.currentReading = currentReading;
    }

    public double getVoltage() {
        return voltage;
    }

    public double getCurrentReading() {
        return currentReading;
    }

    public void setVoltage(double voltage) {
        this.voltage = voltage;
    }

    public void setCurrentReading(double currentReading) {
        this.currentReading = currentReading;
    }

    public void calibrate(double newReading) {
        this.currentReading = newReading;
        System.out.println("  [Calibrated] " + getName() + " -> New Reading: " + newReading);
    }

    @Override
    public void displaySpecs() {
        super.displaySpecs();
        System.out.printf("  Voltage       : %.1f V%n", voltage);
        System.out.printf("  Reading       : %.2f%n", currentReading);
    }
}

File: python/electronic_sensor.py
from industrial_asset import IndustrialAsset

class ElectronicSensor(IndustrialAsset):

    def __init__(self, name, serial_number, base_value, voltage, current_reading):
        super().__init__(name, serial_number, base_value)   # reuse parent __init__
        self.__voltage         = voltage
        self.__current_reading = current_reading

    @property
    def voltage(self):
        return self.__voltage

    @voltage.setter
    def voltage(self, value):
        self.__voltage = value

    @property
    def current_reading(self):
        return self.__current_reading

    @current_reading.setter
    def current_reading(self, value):
        self.__current_reading = value

    def calibrate(self, new_reading):
        self.__current_reading = new_reading
        print(f"  [Calibrated] {self.name} -> New Reading: {new_reading}")

    def display_specs(self):
        super().display_specs()
        print(f"  Voltage       : {self.__voltage:.1f} V")
        print(f"  Reading       : {self.__current_reading:.2f}")
